tests/php/test-class-amp-story-post-type.php

```diff
@@ -696,4 +721,98 @@ public function get_gfont_data() {
 			],
 		];
 	}
+
+	public function get_default_settings_definitions() {
+		return [
+			[
+				[
+					'auto_advance_after'          => [
+						'meta_args' => [
+							'type'              => 'string',
+							'sanitize_callback' => function( $value ) {
+								$valid_values = [ '', 'auto', 'time', 'media' ];
+
+								if ( ! in_array( $value, $valid_values, true ) ) {
+									return '';
+								}
+								return $value;
+							},
+						],
+						'data'      => [
+							'options' => [
+								[
+									'value'       => '',
+									'label'       => __( 'Manual', 'amp' ),
+									'description' => '',
+								],
+								[
+									'value'       => 'auto',
+									'label'       => __( 'Automatic', 'amp' ),
+									'description' => __( 'Based on the duration of all animated blocks on the page', 'amp' ),
+								],
+								[
+									'value'       => 'time',
+									'label'       => __( 'After a certain time', 'amp' ),
+									'description' => '',
+								],
+								[
+									'value'       => 'media',
+									'label'       => __( 'After media has played', 'amp' ),
+									'description' => __( 'Based on the first media block encountered on the page', 'amp' ),
+								],
+							],
+						],
+					],
+					'auto_advance_after_duration' => [
+						'meta_args' => [
+							'type'              => 'integer',
+							'sanitize_callback' => function( $value ) {
+								$value = intval( $value );
+
+								return filter_var(
+									$value,
+									FILTER_VALIDATE_INT,
+									[
+										'default'   => 0,
+										'min_range' => 1,
+										'max_range' => 100,
+									]
+								);
+							},
+						],
+						'data'      => [],
+					],
+				],
+			],
+		];
+	}
+
+	/**
+	 * Test the definitions return value
+	 *
+	 * @dataProvider get_default_settings_definitions
+	 * @covers AMP_Story_Post_Type::get_stories_settings_meta_definitions
+	 */
+	public function test_get_stories_settings_meta_definitions( $default_definitions ) {
+		$definitions = AMP_Story_Post_Type::get_stories_settings_definitions();
+		$this->assertEquals( $default_definitions, $definitions );
+	}
+
+	/**
+	 * Test the definitions return value
+	 *
+	 * @covers AMP_Story_Post_Type::add_story_settings_meta_to_new_story
+	 */
+	public function test_add_story_settings_meta_to_new_story() {
```

Comment:
Should we also test an existing story (post), so that we can check existing stories were unaffected.

Reply:
Added a test for this.